### PR TITLE
chore: add full NCP CR example

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

**Comment by noama-nv** [comment]
LGTM

File: config/samples/mellanox.com_v1alpha1_nicclusterpolicy.yaml [hunk 1]
```diff
@@ -57,3 +57,40 @@ spec:
           }
         ]
       }
+  sriovDevicePlugin:
```
Comment (noama-nv) [inline config/samples/mellanox.com_v1alpha1_nicclusterpolicy.yaml:60]: @rollandf secondaryNetwork ipoib is also applicable on openshift

Comment (rollandf) [reply config/samples/mellanox.com_v1alpha1_nicclusterpolicy.yaml:60]: Done

File: hack/templates/samples/mellanox.com_v1alpha1_nicclusterpolicy.template [hunk 2]
```diff
@@ -57,3 +57,45 @@ spec:
...
+    version: {{ .NicFeatureDiscovery.Version }}
+  docaTelemetryService:
+    image: {{ .DOCATelemetryService.Image }}
+    repository: {{ .DOCATelemetryService.Repository }}
+    version: {{ .DOCATelemetryService.Version }}
+  secondaryNetwork:
+    ipoib:
+      image: {{ .Ipoib.Image }}
+      repository: {{ .Ipoib.Repository }}
+      version: {{ .Ipoib.Version }}
```
Comment (ykulazhenkov) [inline hack/templates/samples/mellanox.com_v1alpha1_nicclusterpolicy.template:101]: nit: no new line at the end

**Review by ykulazhenkov — DISMISSED** [review]

File: hack/templates/samples/mellanox.com_v1alpha1_nicclusterpolicy.template (hunk 2 above)
Comment (rollandf) [reply hack/templates/samples/mellanox.com_v1alpha1_nicclusterpolicy.template]: Done

**Review by ykulazhenkov — APPROVED** [review]
LGTM

File: example/crs/mellanox.com_v1alpha1_nicclusterpolicy_cr-full.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ##### Note #####
+# This example contains all the components supported as a reference.
+# User should edit the example and keep only the required components.
+#
```
Comment (tobiasgiese) [inline example/crs/mellanox.com_v1alpha1_nicclusterpolicy_cr-full.yaml:18]:
```suggestion

```

nit: do we need this example text or should we rather remove it?

Comment (rollandf) [reply example/crs/mellanox.com_v1alpha1_nicclusterpolicy_cr-full.yaml:18]: This example is a reference, that was asked from QA team to easily get the components versions, taking into account that in the future these versions will not be available in the Helm values.
This configuration probably is not a real life scenario.

**Comment by rollandf** [comment]
`/retest-blackduck_scan

**Review by noama-nv — APPROVED** [review]

File: example/crs/mellanox.com_v1alpha1_nicclusterpolicy_cr-full.yaml [hunk 4]
```diff
@@ -0,0 +1,118 @@
+# Copyright 2020 NVIDIA
```
Comment (e0ne) [inline example/crs/mellanox.com_v1alpha1_nicclusterpolicy_cr-full.yaml:1]: Please, update copyright header as at [1] and fix the year

[1] https://github.com/Mellanox/network-operator/blob/6800708dee1bdd073b31a24660c74fe509b55735/pkg/migrate/migrate.go#L1

**Review by e0ne — CHANGES_REQUESTED** [review]

File: example/crs/mellanox.com_v1alpha1_nicclusterpolicy_cr-full.yaml (hunk 4 above)
Comment (rollandf) [reply example/crs/mellanox.com_v1alpha1_nicclusterpolicy_cr-full.yaml]: Done

**Review by e0ne — APPROVED** [review]

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**